jsonnet/telemeter/client/kubernetes.libsonnet

```
@@ -244,6 +244,22 @@ local securePort = 8443;
                 record: 'cluster:telemetry_selected_series:count',
                 expr: 'max(federate_samples - federate_filtered_samples)',
               },
+              {
+                expr: ||| 
+                  sum by (job,namespace,pod) (
+                    rate(federate_requests_failed_total{job="telemeter-client"}[5m])
+                  ) > 0
```

Alerting on the ratio of failed requests is usually more convenient and I'd go with a conservative threshold (e.g. at least 20%) since the main use case is to report when telemetry push fails repeatedly.

```suggestion
                  sum by (namespace) (
                    rate(federate_requests_failed_total{job="telemeter-client"}[15m])
                  ) /
                  sum by (namespace) (
                    rate(federate_requests_total{job="telemeter-client"}[15m])
                  ) > 0.2
```

---

Ahhh yes! I was planning on extending the range to 15m but I would never have thought about using a ratio, indeed it makes way more sense! I'll try to remember next time!